tests/integration/acp/index/create_test.go

```diff
@@ -22,21 +22,24 @@ func TestACP_IndexCreateWithSeparateRequest_OnCollectionWithPolicy_NoError(t *te
 		Actions: []any{
 
 			testUtils.AddPolicy{
-				Identity:         testUtils.ClientIdentity(1),
-				Policy:           userPolicy,
-				ExpectedPolicyID: "94eb195c0e459aa79e02a1986c7e731c5015721c18a373f2b2a0ed140a04b454",
+				Identity: testUtils.ClientIdentity(1),
+				Policy:   userPolicy,
 			},
 
 			testUtils.SchemaUpdate{
 				Schema: `
 					type Users @policy(
-						id: "94eb195c0e459aa79e02a1986c7e731c5015721c18a373f2b2a0ed140a04b454",
+						id: "{{.Policy0}}",
```

Comment:
suggestion: I think we can make this easier to use and read by removing `Replace` and selecting the value to insert based on the inner string value.

For example:
```
id: "{{.PolicyID.0}}",
``` 
Would be transformed to use `testUtils.NewPolicyIndex(0)`, without the tests having to spend an extra 4 lines to do so.

The same pattern would be fairly easy to extend to other actions, and could even support them automatically if you follow a similar wrapping pattern to that used by `corekv.TxnAction` - this way the `SchemaUpdate` action never needs to complicate itself with substitution logic, and we as maintainers are somewhat guaranteed a consistent result.

Reply:
You leave me wanting to request changes here Shahzad :)

I understand you may be feeling some time pressure, but in some sense #3521 is requesting that this PR gets redone, because the new issue encompasses almost everything within this PR.
 
I really don't think we should be making this a habit, as it calls in to question what the point of the reviewing is if authors start creating new issues to redo the vast majority of the work that they are trying to commit instead of improving within the PR.

Reply:
The current change was done, after a consensus was reached in the standup, people preferred Islam's `template suggestion`. Also was discussed here (you have given a thumbs up on the `Replace` suggestion): https://github.com/sourcenetwork/defradb/pull/3171#discussion_r1814171193

Now that all of that is done, a completely new request is being made, which undoes that entire initial consensus.

Reply:
I understand that it can be frustrating, but the PR has been open for 3 and a half months and new ideas are going to pop out.

Reply:
> I understand that it can be frustrating, but the PR has been open for 3 and a half months and new ideas are going to pop out.

Can you expand on your solution/idea again? It seems that you are wanting me to undo/remove the template approach (is that correct? or you have a suggestion to also keep the template approach and then made that work?)

Reply:
I think it is still a template approach, but instead of passing stuff to the action, the template targets `state`.

So if state is defined as:
```
type state struct {
   ...
   PolicyIDs []string
}
```
The template would be defined as:
```
id: "{{.PolicyIDs.0}}",
```

Reply:
I am leaving this one out of scope of this PR.